src/minimega/container.go

```diff
@@ -1425,67 +1426,81 @@ func containerMountDefaults(fsPath string) error {
 
 // aggressively cleanup container cruff, called by the nuke api
 func containerNuke() {
-	// walk /sys/fs/cgroup/minimega for tasks, killing each one
-	err := filepath.Walk(CGROUP_PATH, func(path string, info os.FileInfo, err error) error {
+	// walk CGROUP_PATH for tasks, killing each one
+	if _, err := os.Stat(CGROUP_PATH); err == nil {
+		err := filepath.Walk(CGROUP_PATH, containerNukeWalker)
 		if err != nil {
-			return nil
+			log.Errorln(err)
 		}
+	}
 
-		log.Debug("walking file: %v", path)
-
-		switch info.Name() {
-		case "tasks":
-			d, err := ioutil.ReadFile(path)
-			pids := strings.Fields(string(d))
-			for _, pid := range pids {
-				log.Debug("found pid: %v", pid)
+	// Allow udev to sync
+	time.Sleep(time.Second * 1)
```

Comment:
Correction. It does still exist. But not without systemd. Our setup won't
have it, and honestly I'm not ready to cross the systemd rubicon until we
absolutely have to.

@floren
On Mar 1, 2016 11:36, "Jon Crussell" <notifications@github.com> wrote:

> In src/minimega/container.go
> <https://github.com/sandia-minimega/minimega/pull/403#discussion_r54612519>
> :
>
> >
> > -		log.Debug("walking file: %v", path)
> > -
> > -		switch info.Name() {
> > -		case "tasks":
> > -			d, err := ioutil.ReadFile(path)
> > -			pids := strings.Fields(string(d))
> > -			for _, pid := range pids {
> > -				log.Debug("found pid: %v", pid)
> > +	// Allow udev to sync
> > +	time.Sleep(time.Second * 1)
>
> udevadm settle?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sandia-minimega/minimega/pull/403/files#r54612519>.
>